### PR TITLE
FB Library Search

### PR DESCRIPTION
**Pull request by Jdochoa**
Added configuration for FB client library.

imagen

**At the moment the configuration applies only to windows and at the application level.** Connection level implementation pending.
The search order of the library is corrected, remaining as follows:

1. Try specific library, according to the FR configuration.
2. Try to load fbembed.dll from the current application location.
3.  Try to load fbclient.dll from the current application location.
4. Try to locate fbclient.dll through the installation registry. It only looks for "DefaultInstance"; If your installation has another name, this is not taken into account.
4.1. Try to locate %FB Directory%\fbclient.dll
4.2.- Try to locate %FB Directory%\wow64\fbclient.dll
4.3.- Try to locate %FB Directory%\bin\fbclient.dll
4.4.- Try to locate %FB Directory%\bin\wow64\fbclient.dll

5.- Let's try from the PATH and System directories for FBClient.dll
6.- Last try : attemps loading gds32.dll from PATH and System directories

Fix: #199 #229 #251 #258

### CODE REVIEW

File: src/gui/DatabaseRegistrationDialog.cpp [hunk 1]
```diff
@@ -84,7 +85,7 @@ void DatabaseRegistrationDialog::createControls()
 
     label_charset = new wxStaticText(getControlsPanel(), -1, _("Charset:"));
     long comboStyle =  wxCB_DROPDOWN;
-#ifndef __WXMAC__
```
Comment (arvanus) [inline src/gui/DatabaseRegistrationDialog.cpp:87]: @Jdochoa do you remember why this change? (just curious because looks like a type

File: src/metadata/domain.cpp [hunk 2]
```diff
@@ -66,7 +66,7 @@ std::string Domain::getLoadStatement(bool list)
             " on l.rdb$collation_id = f.rdb$collation_id"
             " and l.rdb$character_set_id = f.rdb$character_set_id"
         " left outer join rdb$types t on f.rdb$field_type=t.rdb$type"
-        " where t.rdb$field_name='RDB$FIELD_TYPE' and f.rdb$field_name ");
+        " where trim(t.rdb$field_name)='RDB$FIELD_TYPE' and f.rdb$field_name ");
```
Comment (arvanus) [inline src/metadata/domain.cpp:69]: Is this trim necessary? Not sure, but maybe this might degrade performance a little

File: src/gui/DatabaseRegistrationDialog.cpp (hunk 1 above)
Comment (Jdochoa) [reply src/gui/DatabaseRegistrationDialog.cpp:87]: Sorry, it was my mistake. I already corrected it

File: src/metadata/domain.cpp (hunk 2 above)
Comment (Jdochoa) [reply src/metadata/domain.cpp:69]: Yes, in the scenario reported in #260, Firebird has an bug because the rdb$types table does not remove whitespace.

You can verify it with this query:

select *, character_length(rdb$field_name)
from rdb$types

Comment (arvanus) [reply src/metadata/domain.cpp:69]: Too bad, do you know if this bug is reported to Fb?

Comment (Jdochoa) [reply src/metadata/domain.cpp:69]: Hi @arvanus, @livius2 .

Please execute this query in these three scenarios:
select *
from rdb$types
where rdb$field_name = 'RDB$FIELD_TYPE'

1.- In the scenario described in #260: The result is empty. (here is the bug)
2.- In the scenario described in #260 but with CharSet UTF8. Returns the list of columns
3.- In any other base without collation like #260 Returns the list of columns

The solution is:

select *
from rdb$types
where trimr(db$field_name) = 'RDB$FIELD_TYPE'

or

select *
from rdb$types
where rdb$field_name like 'RDB$FIELD_TYPE%'

Comment (arvanus) [reply src/metadata/domain.cpp:69]: @Jdochoa I believe this is a Firebird bug, looks like `NO PAD` option from the collation are affecting incorrectly `RDB$` tables
I suggest to don't "fix" this right now without understanding better the situation

Comment (arvanus) [reply src/metadata/domain.cpp:69]: @Jdochoa  I've found another way to fix it, if you pass `RDB$FIELD_TYPE` as a parameter, it'll work correctly.

Looks like the method `std::string Domain::getLoadStatement(bool list)` is used at rows 96 and 450
I can pull a fix if you see no problem, just revert the trim first please

**Review by arvanus — APPROVED** [review]

File: src/metadata/domain.cpp (hunk 2 above)
Comment (arvanus) [reply src/metadata/domain.cpp:69]: I'll accept the pull and fix here

**Comment by arvanus** [comment]
@Jdochoa I think you need to separate this config, one path for x64 and other for x86 build, and this should be transparent for the user.
Probably need to add a config for flamerobin config manager to show/hide configurations according to it's plataform
